src/ui/menu_nbgl.c

```diff
@@ -137,24 +127,38 @@ static void controls_callback(int token, uint8_t index) {
         } else {
             // toggle the switch value
             switch_value = !N_storage.dummy2_allowed;
+            switches[DUMMY_SWITCH_2_ID].initState = (nbgl_state_t) switch_value;
             // store the new setting value in NVM
             nvm_write((void*) &N_storage.dummy2_allowed, &switch_value, 1);
         }
     }
 }
 
-// settings menu definition
-void ui_menu_settings() {
-#define TOTAL_SETTINGS_PAGE  (2)
-#define INIT_SETTINGS_PAGE   (0)
-#define DISABLE_SUB_SETTINGS (false)
-    nbgl_useCaseSettings(APPNAME,
-                         INIT_SETTINGS_PAGE,
-                         TOTAL_SETTINGS_PAGE,
-                         DISABLE_SUB_SETTINGS,
-                         ui_menu_main,
-                         nav_callback,
-                         controls_callback);
+// home page definition
+void ui_menu_main(void) {
+
+    // Initialize switches data
+    switches[DUMMY_SWITCH_1_ID].initState = (nbgl_state_t) N_storage.dummy1_allowed;
```

Comment:
Should we provide a intialization function, to help 3rd party devs making sure they init all the fields ?